### ISSUE TITLE
[Bug]: ocm-provider is missing from nextcloud-27.1.2.zip

### ISSUE THREAD

**Issue by simmarn** [nextcloud/server#40920]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I upgraded Nextcloud from 27.1.1 to 27.1.2 using updater.phar. After upgrade I get a warning in admin/overview:
---
There are some warnings regarding your setup.

    Your web server is not properly set up to resolve "/nextcloud/ocm-provider/". This is most likely related to a web server configuration that was not updated to deliver this folder directly. Please compare your configuration against the shipped rewrite rules in ".htaccess" for Apache or the provided one in the documentation for Nginx at it's [documentation page ↗](https://docs.nextcloud.com/server/27/go.php?to=admin-nginx). On Nginx those are typically the lines starting with "location ~" that need an update.
---

It turns out that ocm-provider folder is missing from my /var/www/nextcloud folder. When I check inside the downloaded nextcloud-27.1.2.zip (https://download.nextcloud.com/server/releases/nextcloud-27.1.2.zip), I see that ocm-provider folder is missing.

### Steps to reproduce

1. On Nextcloud 27.1.1 running on Ubuntu 22.04.3 LTS. No issues in admin/overview
2. Upgrade Nextcloud using updater.phar
3. After upgrade ocm-provider folder is missing

### Expected behavior

ocm-folder shall be kept or reinstalled during upgraede

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "updatechecker": false,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.31.211",
            "***REMOVED SENSITIVE VALUE**"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE**",
        "dbtype": "mysql",
        "version": "27.1.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Stockholm",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0,
        "logtype": "owncloud",
        "logfile": "\/media\/ocdata\/nextcloud.log",
        "theme": "",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1125",
        "mysql.utf8mb4": true,
        "app_install_overwrite": [
            "calendar"
        ],
        "updater.release.channel": "stable",
        "default_phone_region": "SE"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.19.0
  - admin_audit: 1.17.0
  - bruteforcesettings: 2.7.0
  - calendar: 4.5.2
  - circles: 27.0.1
  - cloud_federation_api: 1.10.0
  - comments: 1.17.0
  - contacts: 5.4.2
  - contactsinteraction: 1.8.0
  - dashboard: 7.7.0
  - dav: 1.27.0
  - federatedfilesharing: 1.17.0
  - federation: 1.17.0
  - files: 1.22.0
  - files_pdfviewer: 2.8.0
  - files_reminders: 1.0.0
  - files_rightclick: 1.6.0
  - files_sharing: 1.19.0
  - files_trashbin: 1.17.0
  - files_versions: 1.20.0
  - firstrunwizard: 2.16.0
  - logreader: 2.12.0
  - lookup_server_connector: 1.15.0
  - mail: 3.4.2
  - notes: 4.8.1
  - notifications: 2.15.0
  - oauth2: 1.15.1
  - password_policy: 1.17.0
  - photos: 2.3.0
  - privacy: 1.11.0
  - provisioning_api: 1.17.0
  - qownnotesapi: 23.6.0
  - recommendations: 1.6.0
  - related_resources: 1.2.0
  - serverinfo: 1.17.0
  - settings: 1.9.0
  - sharebymail: 1.17.0
  - support: 1.10.0
  - survey_client: 1.15.0
  - suspicious_login: 5.0.0
  - systemtags: 1.17.0
  - text: 3.8.0
  - theming: 2.2.0
  - twofactor_backupcodes: 1.16.0
  - updatenotification: 1.17.0
  - user_status: 1.7.0
  - viewer: 2.1.0
  - weather_status: 1.7.0
  - workflowengine: 2.9.0
Disabled:
  - encryption: 2.15.0
  - files_external: 1.19.0
  - files_markdown: 2.4.1 (installed 2.4.1)
  - nextcloud_announcements: 1.16.0 (installed 1.5.0)
  - twofactor_totp: 9.0.0
  - user_ldap: 1.17.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by solracsf** [comment]
This is expected. https://github.com/nextcloud/server/pull/40586
You need to fix your webserver.

**Comment by simmarn** [comment]
Fix how? The admin/overview warning says that webserver is not set up to resolve ocm-provider. How can I set up apache2 to resolve ocm-provider when it is removed?

**Comment by chlarsen** [comment]
There is a simple reason for this error, as described here: [here](https://help.nextcloud.com/t/with-nextcloud-update-not-properly-set-up-to-resolve-ocm-provider-and-ocs-provider-on-nginx-server/158528/23?u=chlarsen). Since Nextcloud version 27.1.2, there is no more `/ocm-provider` folder. Looks like an obsolete error check plus message to me, unless I got it all wrong!

**Comment by joshtrichards** [comment]
Is your installation in a subfolder (e.g. `https://domain.com/nextcloud` rather than `https://domain.com/`)? If so, it may need the fix pending in #40843/#40856 for the bundled `.htaccess` file. It'll be released in v27.1.3.

You can manually fix it for now by removing the slash in front of `index.php` as shown here:

https://github.com/nextcloud/server/pull/40856/files

**Comment by joshtrichards** [comment]
Fixed in #40843

**Comment by dd5fz** [comment]
Reposting here, because it might fit better:

https://github.com/nextcloud/server/issues/40803#issuecomment-1833825773
